hbase-backup/src/test/java/org/apache/hadoop/hbase/backup/TestIncrementalBackupWithContinuous.java

```diff
@@ -0,0 +1,254 @@
+      String backup2 = backupTables(BackupType.INCREMENTAL, List.of(table1), BACKUP_ROOT_DIR, true);
+      assertTrue(checkSucceeded(backup2));
+      assertEquals(expectedRowCount, TEST_UTIL.countRows(table1));
+      assertEquals(0, systemTable.readBulkloadRows(List.of(table1)).size());
+      int rowCountAfterBackup2 = expectedRowCount;
+
+      // Doing another bulk load, to check that this data will disappear after a restore operation
+      performBulkLoad("bulk7", methodName);
+      expectedRowCount += ROWS_IN_BULK_LOAD;
+      assertEquals(expectedRowCount, TEST_UTIL.countRows(table1));
+      List<BulkLoad> bulkloadsTemp = systemTable.readBulkloadRows(List.of(table1));
+      assertEquals(1, bulkloadsTemp.size());
+      BulkLoad bulk7 = bulkloadsTemp.get(0);
+
+      // Doing a restore. Overwriting the table implies clearing the bulk loads,
+      // but the loading of restored data involves loading bulk data, we expect 2 bulk loads
+      // associated with backup 3 (loading of full backup, loading of incremental backup).
+      BackupAdmin client = getBackupAdmin();
+      client.restore(BackupUtils.createRestoreRequest(BACKUP_ROOT_DIR, backup2, false,
+        new TableName[] { table1 }, new TableName[] { table1 }, true));
+      assertEquals(rowCountAfterBackup2, TEST_UTIL.countRows(table1));
+      List<BulkLoad> bulkLoads = systemTable.readBulkloadRows(List.of(table1));
+      assertEquals(3, bulkLoads.size());
+      conf1.setBoolean(REPLICATION_MARKER_ENABLED_KEY, REPLICATION_MARKER_ENABLED_DEFAULT);
+    }
+  }
+
+  private void verifyTable(Table t1) throws IOException {
+    Get g = new Get(ROW);
+    Result r = t1.get(g);
+    assertEquals(1, r.size());
+    assertTrue(CellUtil.matchingQualifier(r.rawCells()[0], COLUMN));
+  }
+
+  private void performBulkLoad(String keyPrefix, String testDir) throws IOException {
+    FileSystem fs = TEST_UTIL.getTestFileSystem();
+    Path baseDirectory = TEST_UTIL.getDataTestDirOnTestFS(testDir);
+    Path hfilePath =
+      new Path(baseDirectory, Bytes.toString(famName) + Path.SEPARATOR + "hfile_" + keyPrefix);
+
+    HFileTestUtil.createHFile(TEST_UTIL.getConfiguration(), fs, hfilePath, famName, qualName,
+      Bytes.toBytes(keyPrefix), Bytes.toBytes(keyPrefix + "z"), ROWS_IN_BULK_LOAD);
+
+    listFiles(fs, baseDirectory, baseDirectory);
+
+    Map<BulkLoadHFiles.LoadQueueItem, ByteBuffer> result =
+      BulkLoadHFiles.create(TEST_UTIL.getConfiguration()).bulkLoad(table1, baseDirectory);
+    assertFalse(result.isEmpty());
+  }
+
+  private static Set<String> listFiles(final FileSystem fs, final Path root, final Path dir)
+    throws IOException {
+    Set<String> files = new HashSet<>();
+    FileStatus[] list = CommonFSUtils.listStatus(fs, dir);
+    if (list != null) {
+      for (FileStatus fstat : list) {
+        if (fstat.isDirectory()) {
+          LOG.info("Found directory {}", Objects.toString(fstat.getPath()));
```

Comment:
nit:  You can just use `fstat.getPath()` here.

```suggestion
          LOG.info("Found directory {}", fstat.getPath());
```